resources/keycloak/custom-user-provider/src/main/java/com/cvmanager/auth/provider/user/CustomUserStorageProvider.java

```diff
@@ -0,0 +1,315 @@
+/**
+ * 
+ */
+package com.cvmanager.auth.provider.user;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import org.keycloak.component.ComponentModel;
+import org.keycloak.models.GroupModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.storage.StorageId;
+import org.keycloak.storage.UserStorageProvider;
+import org.keycloak.storage.user.UserLookupProvider;
+import org.keycloak.storage.user.UserQueryProvider;
+import org.keycloak.storage.user.UserRegistrationProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import static com.cvmanager.auth.provider.user.CustomUserStorageProviderConstants.*;
+
+import com.cvmanager.auth.provider.RuntimeStorageException;
+import com.cvmanager.auth.provider.user.pojos.UserObject;
+
+public class CustomUserStorageProvider implements UserStorageProvider,
+        UserLookupProvider,
+        UserQueryProvider,
+        UserRegistrationProvider {
+
+    private static final Logger log = LoggerFactory.getLogger(CustomUserStorageProvider.class);
+    private KeycloakSession ksession;
+    private ComponentModel model;
```

Comment:
suggestion(non-blocking):  By making fields final, you are reducing the number of "moving parts" readers of your code have to think about, making it much easier to reason about the code.
```suggestion
    private final KeycloakSession ksession;
    private final ComponentModel model;
```

Reply:
done